### DISCUSSION TITLE
查询返回结果中小数点后面的值被自动截取了

### DISCUSSION THREAD

**Discussion by juice411** [apache/iotdb#3816] [Q&A] [answered]
如图：
![cfa3c8ac224601cded0592a4d2f20bc](https://user-images.githubusercontent.com/10968514/127091639-6229ac08-70a3-43d7-aded-219535238db9.png)
last 查询结果为：2805.56055
select 查询结果为：2805.56
后面的数字被截取了

**Comment by HTHou** [comment]
show timeseries看看序列的详细信息？

**Comment by juice411** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/iotdb"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月27日(星期二) 晚上7:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [apache/iotdb] 查询返回结果中小数点后面的值被自动截取了 (#3630)





 
show timeseries看看序列的详细信息？

**Comment by juice411** [comment]
![image](https://user-images.githubusercontent.com/10968514/127148467-01254fbf-8e41-45aa-84ca-b7d6a8fd6759.png)

**Comment by HTHou** [comment]
这是因为使用了 RLE编码方式对于浮点数是有损编码导致的，如果需要保证精度，对于浮点数可以使用 gorilla 编码。

**Comment by juice411** [comment]
如何对线上系统做修改？修改完后影响已归档数据吗

**Comment by qiaojialin** [comment, marked as answer]
改 iotdb-engine.properties 里的这个参数：
float_precision=6
反注释重启生效。不影响已归档数据，只影响新写的数据。

**Comment by juice411** [comment]
这个方案可以，多谢！
